components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManager.java

```diff
@@ -691,7 +691,7 @@ public User updateUserPassword(String code, String password, Property[] properti
         } catch (IdentityRecoveryException e) {
             userRecoveryData = userRecoveryDataStore.load(code);
         }
-        updateIdentityContext(userRecoveryData);
+        enterFlow(userRecoveryData);
```

Comment:
Let's use try with finally in here as well

Reply:
Fixed 70f0aece2d5585238e4d7a027c21503aa8ef5a25